### ISSUE TITLE
ax.secondary_xaxis() results in duplicate ticks if matplotlib.rcParams['xtick.top'] = 'True'

### ISSUE THREAD

**Issue by cgadal** [matplotlib/matplotlib#20125]
### Bug report

**Bug summary**

`ax.secondary_xaxis()` results in duplicated ticks if `matplotlib.rcParams['xtick.top'] = 'True'`

**Code for reproduction**

```python
import matplotlib.pyplot as plt
import matplotlib

matplotlib.rcParams['xtick.top'] = 'True'


x = [1, 2, 3]
y = [1, 2, 3]
fac = 1.2

plt.figure(constrained_layout=True)
plt.scatter(x, y)
secax = plt.gca().secondary_xaxis('top', functions=(lambda x: fac*x, lambda x: x/fac))
plt.show()

```

**Actual outcome**

![bug](https://user-images.githubusercontent.com/48522112/116715877-699ea700-a9d7-11eb-8324-0b29f2807295.png)

**Expected outcome**

I feel it should replace the already present ticks.

**Matplotlib version**

  * Operating system: Ubuntu 20.04
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`): 3.3.4
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.8.5
  * Jupyter version (if applicable): 
  * Other libraries:

**Comment by jklymak** [comment]
I don't agree.  Secondary_axis can be applied anywhere inside the parent (or outside).  We don't know where and we don't have any control over what the user may have added to the parent axes. Keep-it-simple says we let the user toggle the ticks off if they do not want them there.

**Comment by jklymak** [comment]
I'm going to close this because we don't infer what the secondary axis may or may not be plotted over.
